File: Java Files/HUAWEI3.java
import java.io.Console;
import java.util.Scanner;

public class HUAWEI3 {
    static void QandV(Scanner sc1) {
        String answer = "";
        // Scanner sc1=new Scanner(System.in);
        String s = sc1.nextLine();
        int N = Integer.parseInt(s.substring(0, 1));
        int M = Integer.parseInt(s.substring(2, 3));
        String[] commands = new String[M];
        int[] grades = new int[N];
        s = sc1.nextLine();
        for (int i = 0; i < N; i++) {
            grades[i] = Integer.parseInt(s.substring(2 * i, 2 * i + 1));
            // Console.WriteLine(grades[i]);
        }
        for (int i = 0; i < M; i++) {
            commands[i] = sc1.nextLine();
            // Console.WriteLine(grades[i]);
            if (commands[i].charAt(0) == "Q".toCharArray()[0]) {
                int head = Integer.parseInt(commands[i].substring(2, 3));
                int tail = Integer.parseInt(commands[i].substring(4, 5));
                if (head >= tail) {
                    int temp = head;
                    head = tail;
                    tail = temp;
                }
                answer += max(grades, head - 1, tail);
                if (i != M - 1)
                    answer += "\n";
                // System.out.println(max(grades,head-1,tail));
            } else {
                int id = Integer.parseInt(commands[i].substring(2, 3));
                int change = Integer.parseInt(commands[i].substring(4, 5));
                grades[id - 1] = change;
            }
        }
        // System.out.println(answer);
        if(sc1.hasNextInt())
        {
            System.out.println(answer);
            QandV(sc1);
        }
        else
        {
            System.out.println(answer);
            return;
        }
        return;
        // sc1.nextLine();
    }

    static int max(int[] g, int a, int b) {
        int max = 0;
        for (int i = a; i < b; i++) {
            if (g[i] >= max)
                max = g[i];
        }
        return max;
    }

    public static void main(String[] args) {
        Scanner sc1 = new Scanner(System.in);

        QandV(sc1);

        // if ()
        // {

        // int a=0;
        // }

        // Console.ReadLine();
    }

}

File: C# Files/HUAWEI3.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp2
{
    public class HUAWEI3
    {
        static void QandV(String s)
        {
            String answer = "";
            // String s = Console.ReadLine();
            int N = int.Parse(s.Substring(0, 1));
            int M = int.Parse(s.Substring(2, 1));
            if (N == 0 || M == 0)
            {

            }
            else
            {
                String[] commands = new string[M];
                int[] grades = new int[N];
                s = Console.ReadLine();
                for (int i = 0; i < N; i++)
                {
                    grades[i] = int.Parse(s.Substring(i * 2, 1));
                    // Console.WriteLine(grades[i]);
                }
                for (int i = 0; i < M; i++)
                {
                    commands[i] = Console.ReadLine();
                    // Console.WriteLine(grades[i]);
                    if (commands[i][0] == "Q".ToCharArray()[0])
                    {
                        int head = int.Parse(commands[i].Substring(2, 1));
                        int tail = int.Parse(commands[i].Substring(4, 1));
                        if (head >= tail)
                        {
                            int temp = head;
                            head = tail;
                            tail = temp;
                        }
                        // answer += Convert.ToString(max(grades, head - 1, tail));
                        // if (i != M - 1)
                        //     answer += "\n";
                        Console.WriteLine(max(grades, head - 1, tail));

                    }
                    else
                    {
                        int id = int.Parse(commands[i].Substring(2, 1));
                        int change = int.Parse(commands[i].Substring(4, 1));
                        grades[id - 1] = change;
                    }
                }
            }

            // Console.WriteLine(answer);
            s = "";
            s = Console.ReadLine();
            if (s != "")
                QandV(s);
            else
                return;
        }

        static int max(int[] g, int a, int b)
        {
            int max = 0;
            for (int i = a; i < b; i++)
            {
                if (g[i] >= max)
                    max = g[i];
            }
            return max;
        }

        static void test()
        {
            String input = Console.ReadLine();            
            String[] subinputs = input.Split(" ");
            foreach (var subinput in subinputs)
            {
                String[] finalsubs=subinput.Split("\\");
                String tail=finalsubs[finalsubs.Count()];
                String[] nameandline=tail.Split(" ");
            }
                Console.WriteLine(" ");
        }
        static void Main1()
        {
            test();
            Console.ReadLine();
            // Console.ReadLine();
        }
    }
}
